backend/services/implementations/courseService.ts

```diff
@@ -34,7 +35,48 @@ const publishedModulesOnlyQuery = [
 ];
 
 class CourseService implements ICourseService {
+  imageService: IImageUploadService;
+
+  constructor(imageService: IImageUploadService) {
+    this.imageService = imageService;
+  }
+
   /* eslint-disable class-methods-use-this */
+  private async hydrateImages(
+    course: CourseResponseDTO,
+  ): Promise<CourseResponseDTO> {
+    const coursePreviewImage = course.image
+      ? await this.imageService.download(course.image)
+      : course.previewImage;
+    const newModules = await Promise.all(
+      course.modules.map(async (module) => {
+        const modulePreviewImage = module?.image
+          ? await this.imageService.download(module.image)
+          : module?.previewImage;
```

Comment:
I know when I made the data model, I had put in a module `image` and `previewImage` property. Obviously I've never made use of them, but is there a meaningful distinction between the two? It seems like previewImage is being used as a backup?